lib/srv/discovery/discovery.go

```diff
@@ -603,6 +631,84 @@ func (s *Server) handleAzureDiscovery() {
 	}
 }
 
+func (s *Server) filterExistingGCPNodes(instances *server.GCPInstances) {
+	nodes := s.nodeWatcher.GetNodes(s.ctx, func(n services.Node) bool {
+		labels := n.GetAllLabels()
+		_, projectIDOK := labels[types.ProjectIDLabel]
+		_, zoneOK := labels[types.ZoneLabel]
+		_, nameOK := labels[types.NameLabel]
+		return projectIDOK && zoneOK && nameOK
+	})
+	var filtered []*gcp.Instance
+outer:
+	for _, inst := range instances.Instances {
+		for _, node := range nodes {
+			match := types.MatchLabels(node, map[string]string{
+				types.ProjectIDLabel: inst.ProjectID,
+				types.ZoneLabel:      inst.Zone,
+				types.NameLabel:      inst.Name,
+			})
+			if match {
+				continue outer
+			}
+		}
+		filtered = append(filtered, inst)
+	}
+	instances.Instances = filtered
+}
+
+func (s *Server) handleGCPInstances(instances *server.GCPInstances) error {
+	client, err := s.Clients.GetGCPInstancesClient(s.ctx)
+	if err != nil {
+		return trace.Wrap(err)
+	}
+	s.filterExistingGCPNodes(instances)
+	if len(instances.Instances) == 0 {
+		return trace.Wrap(errNoInstances)
```

Comment:
```suggestion
		return trace.NotFound(errNoInstances)
```

Reply:
`errNoInstances` is already an error type, so we can't wrap it with `trace.NotFound`.